controllers/datafoundationclient.go

```diff
@@ -125,6 +136,9 @@ func (r *dataFoundationClientReconciler) reconcilePhases() (ctrl.Result, error)
 	if !r.offering.DeletionTimestamp.IsZero() {
 		return ctrl.Result{}, nil
 	} else {
+		if err := r.reconcileWatchers(); err != nil {
+			return ctrl.Result{}, err
+		}
```

Comment:
The code in this func needs to run before anything else in the reconcilePhases func as it is an assumption that we need to ensure before having any reconcile code.